semver/less_than_range.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+import type { Comparator, Range, SemVer } from "./types.ts";
+import { testComparatorSet } from "./_test_comparator_set.ts";
+import { isWildcardComparator } from "./_shared.ts";
+import { compare } from "./compare.ts";
+
+/** Check if the semver is less than the range. */
+export function lessThanRange(semver: SemVer, range: Range): boolean {
+  for (const comparatorSet of range) {
+    if (comparatorSet.some((comparator) => isWildcardComparator(comparator))) {
+      return false;
+    }
+    // If the semver satisfies the comparator set, then it's not less than the range.
+    if (testComparatorSet(semver, comparatorSet)) {
+      return false;
+    }
+    // If the semver is greater than any of the comparator set, then it's not less than the range.
+    if (
+      comparatorSet.some((comparator) =>
+        greaterThanComparator(semver, comparator)
+      )
+    ) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function greaterThanComparator(
```

Comment:
Nit: This is more of a personal preference, but I like it when functions are defined before they're used. In other words, I'd prefer this function be moved before `lessThanRange()`.